memecity.engine.ecs/EntityManager.h

```diff
@@ -1,54 +1,137 @@
 #ifndef _ENTITY_MANAGER_H
 #define  _ENTITY_MANAGER_H
-#include <iostream> 
-#include <map> 
-#include <iterator> 
+#include <iostream>
+#include <iterator>
 #include <vector>
+#include <map>
 #include "Entity.h"
 #include "Component.h"
 #include "System.h"
-using namespace std;
 
-class EntityManager {
-private:
-	int id = 0;
-	vector<Entity*> entities;
-	map<string, vector<Component*>> components;
-	map<string, System*> systems;
+namespace ecs {
+	class EntityManager {
+	private:
+		int last_id = -1;
+		std::vector<Entity> entities;
+		std::map<component_typetoken, std::vector<std::unique_ptr<Component>>> components;
+		std::map<system_typetoken, std::unique_ptr<System>> systems;
 
-public:
-	EntityManager();
-	~EntityManager();
+	public:
+		EntityManager() {}
 
-	///<summary>Creates a new entity with an unused ID.</summary>
-	Entity* create_entity();
+		///<summary>Creates a new entity with an unused ID.</summary>
+		Entity& create_entity()
```

Comment:
Implementation of function in header file?

Reply:
because most of the methods are templates, and templating is tedious outside of the header file, I figured might as well put everything in the header file. This can be changed if needed, but is not neccessary right now.